File: SudokuSolver_Swift/SudokuSolver_Swift/logic/simple/NakedSingles.swift
/*
 * http://www.sudokuwiki.org/Getting_Started
 *
 * If an unsolved cell has exactly one candidate, then the candidate must be placed in that cell.
 */
func nakedSingles(board: Board<Cell>) -> [BoardModification] {
    board.cells
        .unsolvedCells
        .filter { $0.candidates.count == 1 }
        .map { .setValue(SetValue(cell: $0, value: $0.candidates.first!)) }
}

File: SudokuSolver_Java/src/main/sudokusolver/java/logic/simple/NakedSingles.java
package sudokusolver.java.logic.simple;

import sudokusolver.java.Board;
import sudokusolver.java.Cell;
import sudokusolver.java.SetValue;
import sudokusolver.java.UnsolvedCell;

import java.util.List;

/*
 * https://www.sudokuwiki.org/Getting_Started
 *
 * If an unsolved cell has exactly one candidate, then the candidate must be placed in that cell.
 */
public class NakedSingles {
    public static List<SetValue> nakedSingles(Board<Cell> board) {
        return board.getCells()
                .stream()
                .filter(UnsolvedCell.class::isInstance)
                .map(UnsolvedCell.class::cast)
                .filter(cell -> cell.candidates().size() == 1)
                .map(cell -> new SetValue(cell, cell.candidates().iterator().next()))
                .toList();
    }
}